File: backend/src/main/java/es/udc/tfg/backend/model/services/TariffService.java
package es.udc.tfg.backend.model.services;

import java.util.List;

import es.udc.tfg.backend.model.common.exceptions.DuplicateInstanceException;
import es.udc.tfg.backend.model.common.exceptions.InstanceNotFoundException;
import es.udc.tfg.backend.model.entities.Tariff;

public interface TariffService {

	Tariff addTariff(Long hotelId, Tariff tariff) throws DuplicateInstanceException, InstanceNotFoundException;

	Tariff updateTariff(Long hotelId, Tariff tariff) throws InstanceNotFoundException, PermissionException;

	void removeTariff(Long hotelId, Long tariffId) throws InstanceNotFoundException, PermissionException;

	Tariff findTariffById(Long id) throws InstanceNotFoundException;

	List<Tariff> findAllTariffs();
}

File: frontend/src/backend/tariffService.js
import {config, appFetch } from './appFetch';


export const addTariff = (tariff, onSuccess, onErrors) => 
    appFetch('/tariffs/addTariff', config('POST', tariff), 
    onSuccess, onErrors);

export const findAllTariffs = (onSuccess,onErrors) => 
    appFetch('/tariffs/tariffs', config('GET'),
    onSuccess, onErrors);

export const updateTariff = (tariff, onSuccess, onErrors) =>
    appFetch(`/tariffs/${tariff.id}`, config('PUT', tariff),
    onSuccess, onErrors);

export const removeTariff = (tariff, onSuccess, onErrors) =>{
    appFetch(`/tariffs/${tariff.id}`, config('DELETE'), 
    onSuccess, onErrors);}

export const findTariffById = (id, onSuccess, onErrors) =>{
    appFetch(`/tariffs/${id}`, config('GET'), 
    onSuccess, onErrors);}